pkg/controllers/infracluster/infracluster_controller_test.go

```diff
@@ -0,0 +1,234 @@
+		Context("When the InfraCluster is Ready", func() {
+			BeforeEach(func() {
+				Expect(cl.Create(ctx, infraClusterWithExternallyManagedByAnnotationWithValue.DeepCopy())).To(Succeed())
+				mustPatchAWSInfraClusterReadiness(infraClusterWithExternallyManagedByAnnotationWithValue.DeepCopy(), true)
+			})
+
+			It("should not change the Status.Ready field", func() {
+				Eventually(komega.Object(bareInfraCluster)).Should(SatisfyAll(
+					HaveField("Status.Ready", BeTrue()),
+					HaveField("Annotations", HaveKeyWithValue(clusterv1.ManagedByAnnotation, thirdPartyAnnotation)),
+				))
+			})
+		})
+		Context("When the InfraCluster is not Ready", func() {
+			BeforeEach(func() {
+				Expect(cl.Create(ctx, infraClusterWithExternallyManagedByAnnotationWithValue.DeepCopy())).To(Succeed())
+				mustPatchAWSInfraClusterReadiness(infraClusterWithExternallyManagedByAnnotationWithValue.DeepCopy(), false)
+			})
+
+			It("should not change the Status.Ready field", func() {
+				Eventually(komega.Object(bareInfraCluster)).Should(SatisfyAll(
+					HaveField("Status.Ready", BeFalse()),
+					HaveField("Annotations", HaveKeyWithValue(clusterv1.ManagedByAnnotation, thirdPartyAnnotation)),
+				))
+			})
+		})
+	})
+
+	Context("When there is an InfraCluster with an externally ManagedBy Annotation with cluster-capi-operator value", func() {
+		Context("When the InfraCluster is Ready", func() {
+			BeforeEach(func() {
+				Expect(cl.Create(ctx, infraClusterWithExternallyManagedByAnnotation.DeepCopy())).To(Succeed())
+				mustPatchAWSInfraClusterReadiness(infraClusterWithExternallyManagedByAnnotation.DeepCopy(), true)
+			})
+
+			It("should not change the Status.Ready field", func() {
+				Eventually(komega.Object(bareInfraCluster)).Should(SatisfyAll(
+					HaveField("Status.Ready", BeTrue()),
+					HaveField("Annotations", HaveKeyWithValue(clusterv1.ManagedByAnnotation, managedByAnnotationValueClusterCAPIOperator)),
+				))
+			})
+		})
+		Context("When the InfraCluster is not Ready", func() {
+			BeforeEach(func() {
+				Expect(cl.Create(ctx, infraClusterWithExternallyManagedByAnnotation.DeepCopy())).To(Succeed())
+				mustPatchAWSInfraClusterReadiness(infraClusterWithExternallyManagedByAnnotation.DeepCopy(), true)
+			})
+
+			It("should change the Status.Ready field to true", func() {
+				Eventually(komega.Object(bareInfraCluster)).Should(SatisfyAll(
+					HaveField("Status.Ready", BeTrue()),
+					HaveField("Annotations", HaveKeyWithValue(clusterv1.ManagedByAnnotation, managedByAnnotationValueClusterCAPIOperator)),
+				))
+			})
+		})
+	})
+})
+
+func mustPatchAWSInfraClusterReadiness(awsInfraCluster *awsv1.AWSCluster, readiness bool) {
+	patchBase := client.MergeFrom(awsInfraCluster.DeepCopy())
+	awsInfraCluster.Status = awsv1.AWSClusterStatus{
+		Ready: readiness,
+	}
+	Expect(cl.Status().Patch(ctx, awsInfraCluster, patchBase)).To(Succeed(), "should have been able to patch AWSCluster")
```

Comment:
What about using eventually and the komega UpdateStatus concept, would that be more stable here or is patch stable enough?

Reply:
Yeah, that's better 👍 will change